independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java

```diff
@@ -87,13 +88,23 @@ public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws
                     continue;
                 }
                 if (c.getBom().getArtifactId().equals("quarkus-bom") || !sawFirstPlatform) {
-                    mainPlatform = c;
+                    mainCatalog = c;
                     sawFirstPlatform = true;
                 }
                 platformBoms.add(c.getBom());
             }
+        } else if (ElementCatalogBuilder.hasElementCatalog(mainCatalog)) {
```

Comment:
method? :)

Reply:
Yes, it's checking whether there is an "element" catalog attached to the `mainCatalog`. A bit of a hack but that's how it is for now.

Reply:
I meant, it would be better to move this whole block to a specific method.

But I know you love those long methods 🤪